Add `tabIndex` prop to React `Icon` component (#849)
primer/octicons: docs/content/packages/react.mdx

)
```

### `focusable`

> You can add the `focusable` attribute to the SVG element via the `focusable` prop, which can be either `true`, `false`, or `auto`.

Reviewer: Is the intent with `focusable="true"` for the `<svg>` to be interactive/be able to receive focus? If so, how would you feel about the prop being exposed here being `tabIndex` and that would be the signal for `focusable` being `true` or `false`?

In other words:

```tsx
<Icon />                                // <svg aria-hidden="true" focusable="false">
<Icon aria-label="test" />              // <svg aria-label="test" role="img" focusable="false">
<Icon aria-label="test" tabIndex={0} /> // <svg aria-label="test" role="img" tabindex="0" focusable="true">
```

If I understand right, `focusable` is only implemented in IE11 and older versions of Edge and so may not function as expected in all browsers if the intent is for the svg to be focusable in that situation.

Most of this is just based on: https://allyjs.io/tutorials/focusing-in-svg.html. Let me know if I'm misunderstanding anything!

)
```

### `tabIndex`

You can add the `tabindex` attribute to an SVG element via the `tabIndex` prop if the SVG element is intended to be interactive.
`tabIndex` prop also controls the `focusable` attribute of the SVG element which is defined by SVG Tiny 1.2 and only implemented in
Internet Explorer and Microsoft Edge.

> If there is no `tabIndex` prop is present (default behaviour), it will set `focusable` attribute to `false` and this is particularly helpful

Reviewer 2: ```suggestion
If there is no `tabIndex` prop is present (default behavior), it will set the `focusable` attribute to `false`. This is helpful
```

for preventing the decorative SVG from being announced by some specialized assistive technology browsing modes which can get delayed
while trying to parse the SVG markup.

```js
// Example usage
import {PlusIcon} from '@primer/octicons-react'
export default () => (
  <PlusIcon aria-label="Interactive Plus Icon" tabIndex={0} /> New Item
)

> ```

Reviewer 3: Clever!
